src/test/java/seedu/address/model/recipe/IngredientTest.java

```diff
@@ -1,119 +1,146 @@
+    }
+
+    @Test
+    public void constructor_invalidEmptyIngredientName_throwsIllegalArgumentException() {
+        String invalidIngredientName = "";
+        double validIngredientQty = 5.5;
+        String validIngredientQuantifier = "g";
+        assertThrows(IllegalArgumentException.class, () ->
+                new Ingredient(invalidIngredientName, validIngredientQty, validIngredientQuantifier));
+    }
+
+    // @Test
+    // public void constructor_nullIngredientQty_throwsIllegalArgumentException() {
+    //     String invalidIngredientName = "Spaghetti";
+    //     double nullIngredientQty = null;
+    //     String validIngredientQuantifier = "g";
+    //     assertThrows(IllegalArgumentException.class,
+    //             () -> new Ingredient(invalidIngredientName, nullIngredientQty, validIngredientQuantifier));
+    // }
+
+    @Test
+    public void constructor_invalidIngredientQty_throwsIllegalArgumentException() {
+        String validIngredientName = "Spaghetti";
+        double invalidIngredientQty = 0;
+        String validIngredientQuantifier = "g";
+        assertThrows(IllegalArgumentException.class, () ->
+                new Ingredient(validIngredientName, invalidIngredientQty, validIngredientQuantifier));
+    }
+
+    @Test
+    public void constructor_nullIngredientQuantifier_throwsIllegalArgumentException() {
+        String invalidIngredientName = "Spaghetti";
+        double validIngredientQty = 5.5;
+        String nullIngredientQuantifier = null;
+        assertEquals("Spaghetti: 5.5",
+                new Ingredient(invalidIngredientName, validIngredientQty, nullIngredientQuantifier).toString());
+    }
+
+    @Test
+    public void constructor_invalidNullIngredientName_throwsIllegalArgumentException() {
+        String nullIngredientName = null;
+        double validIngredientQty = 5.5;
+        String validIngredientQuantifier = "g";
+        assertThrows(NullPointerException.class, () ->
+                new Ingredient(nullIngredientName, validIngredientQty, validIngredientQuantifier));
+    }
+
+    // @Test
+    // public void constructor_invalidNullIngredientQuantifier_throwsIllegalArgumentException() {
+    //     String invalidIngredientName = "Spaghetti";
+    //     double validIngredientQty = 5.5;
+    //     String invalidIngredientQuantifier = null;
+    //     assertThrows(NullPointerException.class,
+    //             () -> new Ingredient(invalidIngredientName, validIngredientQty, invalidIngredientQuantifier));
+    // }
+
+    // @Test
+    // public void constructor_invalidNegativeIngredientQty_throwsIllegalArgumentException() {
+    //     String invalidIngredientName = "Spaghetti";
+    //     String validIngredientQty = "-12.4";
+    //     String validIngredientQuantifier = "g";
+    //     assertThrows(IllegalArgumentException.class,
+    //             () -> new Ingredient(invalidIngredientName, validIngredientQty, validIngredientQuantifier));
+    // }
```

Comment:
Same as previous test, is there a reason why these tests are commented out? or just an oopsie 👀